src/main/java/com/sap/cloudfoundry/client/facade/adapters/CloudFoundryClientFactory.java

```diff
@@ -54,8 +58,17 @@ private ConnectionContext createConnectionContext(String controllerApiHost) {
         getConnectTimeout().ifPresent(builder::connectTimeout);
         getConnectionPoolSize().ifPresent(builder::connectionPoolSize);
         getThreadPoolSize().ifPresent(builder::threadPoolSize);
-        builder.additionalHttpClientConfiguration(client -> client.metrics(true));
+        builder.additionalHttpClientConfiguration(this::getAdditionalHttpClientConfiguration);
         return builder.build();
     }
 
+    private HttpClient getAdditionalHttpClientConfiguration(HttpClient client) {
+        HttpClient clientWithOptions = client;
+        if (getResponseTimeout().isPresent()) {
```

Comment:
what do you think about using `getResponseTimeout.ifPresent(timeout -> client = client.responseTmeout(timeout));` or something similar?

Reply:
I cannot use this because you can't assign non final variables in lambda expressions.